### PR TITLE
Fix argsort inside numba.jit using kind='mergesort'

### PR DESCRIPTION
**Pull request by dachengx**
**What is the problem / what does the code in this PR do**
As the title described. 

This PR is trying to prevent issues like https://github.com/XENONnT/straxen/issues/1175. 

**Can you briefly describe how it works?**

Similar to https://github.com/XENONnT/straxen/pull/1176. 

**Can you give a minimal working example (or illustrate with a figure)?**

Please include the following if applicable:
  - Update the docstring(s)
  - Update the documentation
  - Tests to check the (new) code is working as desired.
  - Does it solve one of the open issues on github?

Please make sure that all automated tests have passed before asking for a review (you can save the PR as a draft otherwise).

### CODE REVIEW

File: strax/run_selection.py [hunk 1]
```diff
@@ -306,7 +306,7 @@ def define_run(self: strax.Context,
     if isinstance(data, (pd.DataFrame, np.ndarray)):
         if isinstance(data, np.ndarray):
             data = pd.DataFrame.from_records(data)
-
```
Comment (dachengx) [inline strax/run_selection.py:309]: This is just to remove an unused line.

**Review by jmosbacher — APPROVED** [review]
Nice catch.
Just wondering  if maybe it would be more future-proof if you had the `_sort_by_time_and_channel` function accept `sort_kind` as an argument so users could choose other options if they know what they are doing.

**Comment by dachengx** [comment]
> Nice catch. Just wondering if maybe it would be more future-proof if you had the `_sort_by_time_and_channel` function accept `sort_kind` as an argument so users could choose other options if they know what they are doing.

Nice suggestion. Updated.

**Comment by jmosbacher** [comment]
Maybe only slightly related to this PR, but would be nice at some point to introduce tests that check that all jit compiled functions produce the same results as their pure python implementations

**Pull request merged**